graph.go

```diff
@@ -149,6 +149,29 @@ func (g *Graph) Node(id string) Node {
 	return n
 }
 
+// DeleteNode deletes a node and all the edges associated to the node
+func (g *Graph) DeleteNode(id string) error {
+	if _, ok := g.findNode(id); ok {
+		// Remove Node
+		delete(g.nodes, id)
+		// Remove all the edges from the Node
+		delete(g.edgesFrom, id)
+		fmt.Printf("%+v", g.edgesFrom)
+		// Remove all the edges to the Node
+		for parent, edgeList := range g.edgesFrom {
+			for i, edge := range edgeList {
+				if edge.to.id == id {
+					g.edgesFrom[parent] = append(g.edgesFrom[parent][:i], g.edgesFrom[parent][i+1:]...)
+				}
+			}
+		}
+
+		return nil
+	} else {
+		return fmt.Errorf("Node %v not found", id)
```

Comment:
i am not sure about returning an error. I am considering using a bool here (just like a Go map delete is doing)